communication/mavsdkvehicleserver.cpp

```diff
@@ -385,6 +385,7 @@ MavsdkVehicleServer::MavsdkVehicleServer(QSharedPointer<VehicleState> vehicleSta
                 heartbeat.type = MAV_TYPE_GROUND_ROVER;
                 heartbeat.autopilot = WAYWISE_MAVLINK_AUTOPILOT_ID;
                 heartbeat.base_mode |= MAV_MODE_FLAG_SAFETY_ARMED | MAV_MODE_FLAG_CUSTOM_MODE_ENABLED; // Note: behave like PX4...
+                heartbeat.custom_mode = static_cast<uint32_t>(mVehicleState->getWaywiseObjectType());
```

Comment:
custom_mode is bitfield, you need to use bitwise operation |= to set the flag

by using = you overwriting the previous flags

Reply:
also we should avoid static cast is not a good practice
we should explicitly declare the enumerator as uint32

enum WAYWISE_OBJECT_TYPE : uint32_t

Reply:
I found another way to send the object type information to the control tower avoiding setting custom mode in the heartbeat which may not be the ideal place for this info. We now provide a parameter from the vehicle server that the control tower can read when instantiating the ground rover. 

IMO, we should not derive WAYWISE_OBJECT_TYPE from uint32_t and do static cast for a specific purpose. This makes it clear to anyone reading the code that there is a need to perform this conversion, and it’s not just an automatic, implicit behavior and use the named enumerators instead of integer values. In most cases, compilers will automatically choose the smallest underlying type that can represent all the enum values. At the end, I think this is a design choice.